py_wtf/cli/__init__.py

```diff
@@ -3,28 +3,35 @@
 # SPDX-License-Identifier: MIT
 from __future__ import annotations
 
+import asyncio
+
 import json
 import logging
 import os
 import shutil
-from functools import partial
+from functools import partial, wraps
 from pathlib import Path
-from typing import Iterable
+
+from typing import Callable, Coroutine, ParamSpec, TypeVar
 
 import click
 import rich
 
 from py_wtf.__about__ import __version__
 from py_wtf.indexer import index_dir, index_file, index_project
 from py_wtf.repository import converter, ProjectRepository
-from py_wtf.types import (
-    Documentation,
-    FQName,
-    Module,
-    Project,
-    ProjectMetadata,
-    ProjectName,
-)
+from py_wtf.types import Documentation, FQName, Project, ProjectMetadata, ProjectName
+
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def coroutine(f: Callable[P, Coroutine[None, None, T]]) -> Callable[P, T]:
```

Comment:
Oh no :D

Reply:
Yeah well this is what the `click` project recommends, so... :)